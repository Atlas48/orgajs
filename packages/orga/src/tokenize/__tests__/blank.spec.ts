import tok from "./tok"

describe("Tokenize Blanks", () => {
  it("knows blank", () => {
    expect(tok("")).toMatchInlineSnapshot(`Array []`)
    expect(tok(" ")).toMatchInlineSnapshot(`Array []`)
    expect(tok("    ")).toMatchInlineSnapshot(`Array []`)
    expect(tok("\t")).toMatchInlineSnapshot(`Array []`)
    expect(tok(" \t")).toMatchInlineSnapshot(`Array []`)
    expect(tok("\t ")).toMatchInlineSnapshot(`Array []`)
    expect(tok(" \t  ")).toMatchInlineSnapshot(`Array []`)
  })

  it("knows these are not blanks", () => {
    expect(tok(" a ")).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "a ",
          "type": "text.plain",
          "value": "a ",
        },
      ]
    `)
  })
})
