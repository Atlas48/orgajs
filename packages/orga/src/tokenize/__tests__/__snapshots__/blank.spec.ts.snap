// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tokenize Blanks knows blank 1`] = `Array []`;

exports[`Tokenize Blanks knows blank 2`] = `Array []`;

exports[`Tokenize Blanks knows blank 3`] = `Array []`;

exports[`Tokenize Blanks knows blank 4`] = `Array []`;

exports[`Tokenize Blanks knows blank 5`] = `Array []`;

exports[`Tokenize Blanks knows blank 6`] = `Array []`;

exports[`Tokenize Blanks knows blank 7`] = `Array []`;

exports[`Tokenize Blanks knows these are not blanks 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 3,
        "line": 0,
      },
      "start": Object {
        "column": 1,
        "line": 0,
      },
    },
    "type": "text.plain",
    "value": "a ",
  },
]
`;
